packages/odyssey-storybook/test/testcafe/framework/pageobjects/buttonPageObject.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {baseUrl} from "../mocks/baseUrl";
+import XPathSelector from "../xpath-selector";
+
+const id = 'components-button'
+const viewMode = 'story'
+const buttonXPathSelector = '//div[@id=\'root\']//Button';
+
+class Button {
+  private readonly type: string;
+  private readonly children: string;
+  private readonly isDisabled: boolean;
+  private readonly size: string;
+  private readonly isWide: boolean;
+  private readonly variant: string;
+
+
+  constructor(
+    type= 'primary',
+    children = 'Button label',
+    isDisabled= false,
+    size= 's',
+    isWide = false,
+    variant = 'primary') {
+    this.type = type;
+    this.children = children;
+    this.isDisabled = isDisabled;
+    this.size = size;
+    this.isWide = isWide;
+    this.variant = variant;
```

Comment:
we might explore using the [TS constructor shorthand](https://blog.logrocket.com/16-useful-typescript-javascript-shorthands-know/#typescript-constructor-shorthand) here at all? I don't recall if the `readonly` modifier is supported using that approach however...

Reply:
I don't think any of these parameters need to be read-only. I had to make them read-only to stop ts complaining. 

The shorthand version seems to be much cleaner.